### PR TITLE
Move Python ucxx code into python/ucxx

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
Also fix up the isort configuration to get more correct results.

Contributes to https://github.com/rapidsai/build-planning/issues/33

### CODE REVIEW

File: python/pyproject.toml
Comment (pentschev) [inline python/pyproject.toml:1]: Why do we need to keep this files if we have one for each of the subprojects `distributed-ucxx` and `ucxx`?

**Review by pentschev — APPROVED** [review]
Assuming there's nothing missing (i.e., CI passes), LGTM. Thanks @KyleFromNVIDIA .

File: python/pyproject.toml
Comment (KyleFromNVIDIA) [reply python/pyproject.toml:1]: Because this is common `isort`, `ruff`, etc. configuration for all of the projects.

**Comment by KyleFromNVIDIA** [comment]
This will probably require a devcontainers change: https://github.com/rapidsai/devcontainers/blob/541895260fa9d3d772989c55c64a3ab98323f1c4/features/src/rapids-build-utils/opt/rapids-build-utils/manifest.yaml#L42

cc @trxcllnt

**Comment by KyleFromNVIDIA** [comment]
I've opened https://github.com/rapidsai/devcontainers/pull/372 for this

File: python/pyproject.toml
Comment (pentschev) [reply python/pyproject.toml:1]: Ah ok, I didn't know you could use `pyproject.toml` as common config for subprojects. Thanks for the reply!

**Review by jameslamb — APPROVED** [review]

**Review by vyasr — APPROVED** [review]

**Pull request merged**